In [10]:
import pandas as pd
import numpy as np
from scipy import stats

#### Q.1. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

Ans : Assume H0 : There is no difference in diameter of the cutlets between two units. 
      H1 : There is significant difference in the diameter of the cutlet between two units.
             
2 Samples so We are using 2 sample test





In [2]:
data = pd.read_csv(r"F:\Data Science\Assignments\3.Hypothesis Testing\Cutlets.csv")

In [4]:
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [8]:
unitA = pd.Series(data.iloc[:, 0])
unitB = pd.Series(data.iloc[:,1])

In [14]:
p_value = stats.ttest_ind(unitA, unitB)


print(p_value[1])

0.4722394724599501


##### p_value = 0.4722394724599501 (> 0.05) Hence, we are accepting Null hypothesis. There is no difference in diameter of the cutlets between two units

#### Question. 2    A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

Ans : Here, we are using ANOVA as number of samples are more than 2.

Assume, H0 : No Variance. All samples TAT means are same.
        H1 : Atleast one sample TAT mean is different


In [16]:
df = pd.read_csv(r"F:\Data Science\Assignments\3.Hypothesis Testing\LabTAT.csv")

In [17]:
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [18]:
result = stats.f_oneway(df.iloc[:,0],df.iloc[:,1],df.iloc[:,2], df.iloc[:,3])

In [19]:
result

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

###### as p_value < 0. so we are rejecting Null hypothesis. 

#### Question 3. Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

Ans : Here, we are determining two categorical values dependent upon each other or not. So we have to use chi2 test 


Assume, 
H0 : groups are independent (males-femals ratio are similar across all regions and does not vary or related to each other)
H1 : Groups are depenedent (males-femakes ration are not similar. it is vary and related to each other)

In [20]:
df = pd.read_csv(r"F:\Data Science\Assignments\3.Hypothesis Testing\BuyerRatio.csv")

In [21]:
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [25]:
array = np.array([[50,142,131,70],[435,1523,1356,750]])
array

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [26]:
stats.chi2_contingency(array)  # output : (stats_value, p_value)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

As p_value = 0.66 is greater than 0.05. We are acceping NULL hypothesis. Groups are independent.males-femals ratio are similar across all regions and does not vary or related to each other

#### Question 4.      TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Ans :

Assume,
H0 : Groups are independent (order form error % are not varies by centre)



In [27]:
form = pd.read_csv(r"F:\Data Science\Assignments\3.Hypothesis Testing\Costomer+OrderForm.csv")

In [28]:
form

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [29]:
form['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [30]:
form['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [31]:
form['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [32]:
form['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [34]:
table = np.array ([[271,267,269,280],[29,33,31,20]])

In [36]:
result = stats.chi2_contingency(table)

In [37]:
result

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

 as p_vale = 0.277 which is greater than 0.05. We are accepting Null Hypothesis. Groups are independent. 
 thus, order form error% does not varies by centre